packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -63,16 +63,23 @@ export abstract class InputMediaDeviceManager<
    * Starts stream.
    */
   async enable() {
-    if (this.state.status === 'enabled') return;
-    this.enablePromise = this.unmuteStream();
-    try {
-      await this.enablePromise;
-      this.state.setStatus('enabled');
-      this.enablePromise = undefined;
-    } catch (error) {
-      this.enablePromise = undefined;
-      throw error;
-    }
+    if (this.state.pendingStatus === 'enabled') return;
+    const signal = this.nextAbortableStatusChangeRequest('enabled');
+    const doEnable = async () => {
+      if (signal.aborted) return;
+      try {
+        await this.unmuteStream();
+        this.state.setStatus('enabled');
+      } finally {
+        if (!signal.aborted) {
+          if (!signal.aborted) this.resetStatusChangeRequest();
+        }
+      }
+    };
+    this.statusChangePromise = this.statusChangePromise
+      ? this.statusChangePromise.then(doEnable)
+      : doEnable();
+    await this.statusChangePromise;
```

Comment:
4. If there's already a pending request in the works, queue to run after it. Otherwise, run immediately.